### PR TITLE
Update the connection data from env on translator creation

### PR DESCRIPTION
**Pull request by wistefan**
## Proposed changes

The change reads out the enviornment variables for connection data on translator connection. This is required, because else the health-check fails until the first real call is made to the api. This leads to a deadlock in k8s environments, since no call is possible before its healthy.

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance (update of libraries or other dependences)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [CLA][cla]
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run **all** the existing tests locally (not just those related to my feature) and there are no errors
- [x] After the last push to the PR branch, I have run the lint script locally and there are no changes to the code base
- [x] I have updated the [RELEASE NOTES][release]
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual_cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/orchestracities/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"
[release]: https://github.com/orchestracities/ngsi-timeseries-api/blob/master/RELEASE_NOTES.md
    "QuantumLeap Release Notes"

### CODE REVIEW

**Comment by wistefan** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/translators/timescale.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ class PostgresTranslator(sql_translator.SQLTranslator):
     def __init__(self, conn_data=PostgresConnectionData()):
         super(PostgresTranslator, self).__init__(
             conn_data.host, conn_data.port, conn_data.db_name)
+        conn_data.read_env()
```
Comment (c0c0n3) [inline src/translators/timescale.py:85]: Is this actually going to work? The connection data gets read after class init! Did you test this? :-) Not a Python expert myself, but maybe we can move this statement before the call to `super`?

On the other hand, if Python moans about super not being the first call in the ctor, then you can zap this statement (i.e. `conn_data.read_env()`) and fix the health endpoint instead. In fact, the PostgressTranslator ctor wasn't meant to be called directly but rather acquired through a [context manager](https://github.com/wistefan/ngsi-timeseries-api/blob/6e2e11d09405ac9488e3dd676ac8a38f49830c8d/src/translators/timescale.py#L314). So the code in the [health endpoint](https://github.com/wistefan/ngsi-timeseries-api/blob/master/src/reporter/health.py#L18-L22) should've rather been

```python
    if db == TIMESCALE_BACKEND:
        from translators.timescale import postgres_translator_instance
        with postgres_translator_instance() as trans:
            health = trans.get_health()
            return health
```

As you've probably noticed our code is quite tangled and is in need of refactoring :-)

**Comment by c0c0n3** [comment]
hi @wistefan and welcome on board! Thanks so much for your contrib! I've done a first review, see my comments there. Basically if you've tested the fix we could keep it---I'm just not sure about the read_env call having the effect you're hoping for, so if you could please double check.

Thanks a million!!

File: src/translators/timescale.py (hunk 1 above)
Comment (chicco785) [reply src/translators/timescale.py:85]: @wistefan  i tend to think this change is not needed, and would like to know why you want to inject it, i.e. what's is the issue you are facing.

Context Manager is used to in the factory, so in the normal processing of incoming requests, configuration is correctly loaded:
https://github.com/orchestracities/ngsi-timeseries-api/blob/master/src/translators/factory.py#L53

Comment (chicco785) [reply src/translators/timescale.py:85]: ok, i got the issue from andrea's comment, so the correct fix is what he proposed. @wistefan can you apply it?

thx

Comment (chicco785) [reply src/translators/timescale.py:85]: with the fix, you won't need `conn_data.read_env()` anymore, can you remove?
many thx

Comment (wistefan) [reply src/translators/timescale.py:85]: Good morning, thanks for the quick reaction. I tested it and it worked. But I prefer your solution, that looks much cleaner. Applied in c39f8225d2876bf68b874eaf867b8f65f477f1ef

Comment (chicco785) [reply src/translators/timescale.py:85]: @wistefan i saw that you applied the solution, but the code in this pr still include `conn_data.read_env()`, can you remove it?

Comment (wistefan) [reply src/translators/timescale.py]: Your right, sorry forgot to push that part. Fixed in 7bcc862656b949888d8f120271caab5b850b1670

**Review by c0c0n3 — APPROVED** [review]

**Pull request merged**